### ISSUE TITLE
Incorrect format for HTML tables

### ISSUE THREAD

**Issue by cderv** [rstudio/rmarkdown#1954]
Here is a minimal reprex
````md
---
title: "table format"
output: html_document
---

```{r}
knitr::kable(iris, "html")
```
````

This is caused by the change in https://github.com/rstudio/rmarkdown/pull/1895 and the fact that `kable` html output does not include odd and even on rows. 

Discovered by a user while loading kableExtra.
````md
---
title: "table format"
output: html_document
---

```{r}
library(knitr)
library(kableExtra)
knitr::kable(iris)
```
```` 

**kableExtra** is modifying the format for `knitr::kable` that default to markdown pipe output. By auto formating to html, just loading the package modify the behavior of kable. 

Two solutions: 
* We could revert and fix another solution for #1893 
* Modify **knitr** html table to include the `odd` and `even` class.

For now the former seems better, as there is a JS solution for #1893 (see https://github.com/rstudio/rmarkdown/issues/1893#issuecomment-688877872)

cc @apreshill

**Comment by cderv** [comment]
In fact this is not a regression. `knitr` html table where not formatted as Pandoc Markdown table because the JS code does not target them. 
With rmarkdown 2.4, before the aforementioned fix I thought was the cause.
![image](https://user-images.githubusercontent.com/6791940/100107043-87387500-2e69-11eb-9197-b4bed3916a15.png)

That means HTML tables from `knitr` where not displayed the same as Pandoc markdown table and the fix in #1893 was not the cause.

This leave us with the question: Do we want knitr html table to be formatted the same as Pandoc mardown table ?

We may also add a test so that we are sure table existing framework are not impacted by our change... or is it something we should wait for revdep check to find out ?

**Comment by cderv** [comment]
Documentation: https://bookdown.org/yihui/rmarkdown-cookbook/kable.html#customize-html-tables

**Comment by yihui** [comment]
> This leave us with the question: Do we want knitr html table to be formatted the same as Pandoc mardown table ?

That would be great. The only tricky thing is that the **knitr** html table doesn't have any special identifier or classes, so it may be hard to reliably detect tables generated by `kable(, 'html')`. Or perhaps we find out tables that do not have any attributes (`id` or `class`) at all and apply the `table table-condensed` classes? That might work.

**Comment by cderv** [comment]
> Or perhaps we find out tables that do not have any attributes (id or class) at all and apply the table table-condensed classes? That might work.

I thought of that, but we would for example target flextable output which have no class or id on `theader` or `tbody` also. I tried it, and the output changed. 

Without adding id or class to knitr output, I guess we need to rely on the `table.attr` as I think there is not really a way for the boostrap style table to target the right table without id or class. 

The previous change we made (from `tr.header` to `tr.odd`) seems more target to pandoc tables.
